### ISSUE TITLE
'use-after-free' warnings when compiling with GCC 12

### ISSUE THREAD

**Issue by mahyquintana** [scipopt/soplex#41]
When SoPLEX is built with GGC 12, the compiler produces 61 warnings like the following one:
`soplex/src/soplex/classset.h:519:17: warning: pointer may be used after 'void free(void*)' [-Wuse-after-free]`

Is this a known issue?
Are these false positive and it's safe to ignore them?

**Comment by leoneifler** [comment]
Yes, this is a known issue and a false positive. The newer compilers complain about this, unfortunately it is not that easy to fix.

**Comment by DominikKamp** [comment]
Fixed with efb397b5f974e88dc094d1e32f4eae3e4be8bbab.
